content/articles/testing-for-broken-authentication-in-web-apps/index.md

```diff
@@ -0,0 +1,117 @@
+
+![http history](/engineering-education/testing-for-broken-authentication-in-web-apps/http.png)
+
+Now in the intruder tab, I'll click on the `positions` tab which displays the intercepted request.
+
+![intruder](/engineering-education/testing-for-broken-authentication-in-web-apps/intruder.png)
+
+I'll leave the `attack type` option as `sniper`. From the screenshot above, the email and password values are highlighted in green. Clicking on the `Clear` button helps remove that.
+
+Next, I'll highlight the password parameter, this allows me to substitute the password value with a list of passwords. To do that I'll click on the `Add` button, click on the `payloads` tab, and under the `Payload Options [Simple List]` i am required to paste a list of passwords that serves as payloads.
+
+I'll be using the [Seclist worst passwords of 2017](https://github.com/danielmiessler/SecLists/blob/master/Passwords/Common-Credentials/worst-passwords-2017-top100-slashdata.txt) as my password list.
+
+I'll copy and click on paste in the payload options section.
+
+![payload](/engineering-education/testing-for-broken-authentication-in-web-apps/payload.png)
+
+After pasting the list of passwords, next is to click on `Start Attack` to begin the attack process.
+
+![attack](/engineering-education/testing-for-broken-authentication-in-web-apps/attack.png)
+
+As soon as the attack process is finished, I scrolled through the result and found a request with a status code of 200 and a payload of `admin123`.
+
+![correct password](/engineering-education/testing-for-broken-authentication-in-web-apps/password.png)
+
+What the attack simply did was send different login requests with each of the passwords in our list.
+
+Now I'd try to log in with the newly discovered password.
+
+![logged in](/engineering-education/testing-for-broken-authentication-in-web-apps/success.png)
+
+Works!
+
+- [Session Fixation Attacks](https://www.netsparker.com/blog/web-security/session-fixation-attacks/)
+This type of vulnerability allows an attacker to hijack a user session.
+
+
+### Impact of Broken Authentication
+- Compromising an account allows the attacker access to unauthorized information.
+- It could lead to full application take over.
+- Loss of sensitive and confidential business information.
+
+### Prevention of Broken Authentication
+- Don't expose sessions IDs in URLs.
+- Don't give room for user enumeration: It's critical to utilize identical, generic error messages and to double-check that they're the same. With every login request, you should return the same HTTP status code.
+- Implement a strong password policy: Allowing the use of weak and well-known passwords is not a good idea. After a given number of login attempts, require users to pass a CAPTCHA test.
+- brute-force protection: prevent brute-force login attempts.
+- Multi-factor authentication: provide an extra layer of security for users.
```

Comment:
Can you add reference links for each of the above methods? This might help the reader to follow up on each of the above said method.